File: solutions/python/bob/2/bob.py
def response(hey_bob):
    hey_bob=hey_bob.strip()
    if not hey_bob: return "Fine. Be that way!"
    elif hey_bob.isupper(): 
        return "Calm down, I know what I'm doing!" if hey_bob[-1]=='?' else "Whoa, chill out!"
    return "Sure." if hey_bob[-1]=='?' else "Whatever."
    
    

File: solutions/csharp/bob/2/Bob.cs
public static class Bob
{
    public static string Response(string statement)
    {
        statement = statement.Trim();
        
        bool isEmpty = string.IsNullOrEmpty(statement);
        bool isQuestion = statement.EndsWith('?');
        bool isYelling = statement.Any(char.IsLetter) && statement == statement.ToUpper();
        
        if (isEmpty) return "Fine. Be that way!";
        if (isQuestion && isYelling) return "Calm down, I know what I'm doing!";
        if (isQuestion) return "Sure.";
        if (isYelling) return "Whoa, chill out!";
        return "Whatever.";
    }
}